src/databricks/sql/utils.py

```diff
@@ -357,7 +357,12 @@ def user_friendly_error_message(self, no_retry_reason, attempt, elapsed):
             user_friendly_error_message = "{}: {}".format(
                 user_friendly_error_message, self.error_message
             )
-        return user_friendly_error_message
+        try:
+            error_context = str(self.error)
+        except:
+            error_context = ""
+
+        return user_friendly_error_message + ". " + error_context
```

Comment:
`str()` may fail for many weird reasons. Maybe let's either set some default message (`error_context = "unknown error"`), or catch the exception and use its message:

```suggestion
        try:
            error_context = str(self.error)
        except  Exception as e:
            error_context = str(e)

        return user_friendly_error_message + ". " + error_context
```

Reply:
a default message makes sense. I'll update.